transports/p2p/p2putil/listeners.go

```diff
@@ -0,0 +1,96 @@
+package p2putil
+
+import (
+	"fmt"
+	"net"
+	"runtime"
+	"strings"
+
+	"github.com/bitcoin-sv/block-headers-service/config"
+	"github.com/rs/zerolog"
+)
+
+// initListeners initializes the configured net listeners and adds any bound
+// addresses to the address manager. Returns the listeners and a NAT interface,
+// which is non-nil if UPnP is in use.
+func InitListeners(log *zerolog.Logger) ([]net.Listener, error) {
```

Comment:
I'm just wondering... Could we store the `log` as a global variable in some module?
That approach would allow us to get rid of such necessity of passing `log` everywhere.